src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/BuildPropertyPage.xaml

```diff
@@ -387,6 +391,45 @@
                DisplayName="When the output is updated" />
   </EnumProperty>
 
+  <DynamicEnumProperty Name="Trimming"
+                DisplayName="Trimming"
+                Description="Select the desired .NET trimming option for optimizing your application's deployment size and performance."
+                HelpUrl="https://learn.microsoft.com/dotnet/core/deploying/native-aot/?tabs=net7"
+                Category="Publish"
+                EnumProvider="TrimmingEnumProvider"
+                MultipleValuesAllowed="False">
+    <DynamicEnumProperty.DataSource>
+      <DataSource HasConfigurationCondition="False"
+                  Persistence="ProjectFileWithInterception"/>
+    </DynamicEnumProperty.DataSource>
+    <DynamicEnumProperty.Metadata>
+      <NameValuePair Name="VisibilityCondition">
+        <NameValuePair.Value>(has-net-core-app-version-or-greater "8.0")</NameValuePair.Value>
+      </NameValuePair>
+    </DynamicEnumProperty.Metadata>
+  </DynamicEnumProperty>
+
+  <BoolProperty Name="PublishAot"
+                DisplayName="Publish native AOT"
+                Description="Enable Ahead-of-Time (AOT) Compilation to generate native machine code for your .NET Core application during the publish process for improved performance and reduced startup time."
+                HelpUrl="https://learn.microsoft.com/dotnet/core/deploying/native-aot/?tabs=net7"
+                Category="Publish">
+    <BoolProperty.DataSource>
+      <DataSource HasConfigurationCondition="False" />
+    </BoolProperty.DataSource>
+    <BoolProperty.Metadata>
+      <NameValuePair Name="VisibilityCondition">
+        <NameValuePair.Value>
+          (and
+            (has-net-core-app-version-or-greater "8.0")
+            (or
+              (not (has-project-capability "WinForms"))
+              (not (has-project-capability "WPF"))))
```

Comment:
I guess we don't re-evaluate visibility conditions when project capabilities change. That seems like something we should support in future. Do you think that would explain the issue you were seeing with this approach?

Using the application page properties seems like a good workaround for now.

Reply:
Oh, that makes sense! Yeah, the application properties did the trick for WPF and WinForms 🙂 
![image](https://github.com/dotnet/project-system/assets/8518253/323c7b64-19f4-4664-afe7-31fc26d76517)

 The interesting part now is to detect when it's a MAUI project or a class library.